tests/checks/test_cross_join_check.py

```diff
@@ -229,3 +229,32 @@ def test_with_cross_join_with_group_and_subquery(self):
         with self.patch_schema(schema):
             stmt = sqlparse.parse(sql)[0]
             assert False == self.has_cross_join(stmt)
+
+    def test_with_no_cross_join_with_limit(self):
+        tbl_a, tbl_b, tbl_c = Mock(), Mock(), Mock()
+        tbl_a.primary_key.columns.keys.return_value = ['arrowid']
+        tbl_b.primary_key.columns.keys.return_value = ['bookid']
+        tbl_c.primary_key.columns.keys.return_value = ['cookid']
+        schema = {
+            'arrow': tbl_a,
+            'book': tbl_b,
+            'cook': tbl_c,
+        }
+        sql = (
+            "SELECT * "
+            "FROM ("
+            "    SELECT arrowid AS objectid"
+            "    FROM arrow"
+            "  UNION ALL"
+            "    SELECT bookid AS objectid"
+            "    FROM book"
+            "  UNION ALL"
+            "    SELECT cookid AS objectid"
+            "    FROM cook"
```

Comment:
This may be a more contrived scenario, but does the following query pass?

```sql
SELECT * FROM (
  SELECT arrowid AS objectid
  FROM arrow, book
  WHERE 1 = 1
  LIMIT 2
UNION ALL
  SELECT cookid AS objectid
  FROM cook
)
ORDER BY objectid DESC
LIMIT 1, 2
```

Reply:
I can add this test.